src/main/java/org/apache/commons/compress/utils/SeekableInMemoryByteChannel.java

```diff
@@ -96,9 +96,7 @@ public long size() {
 
     @Override
     public SeekableByteChannel truncate(long newSize) {
-        if (size > newSize) {
-            size = (int) newSize;
-        }
+        size = (int) newSize;
```

Comment:
The fact that removing the condition made the last test case pass was very interesting but definitely warrants double or even triple-checking. Also I'd check all the test cases again too. Two of us here looked at them for quite a while but I found one mistake in our reasoning while implementing the actual test code already. There might be another.

Reply:
I think removing the size comparion is not right. There're also some problems in your test cases, and I think that's why your code passed the test cases.